lib/modules/datasource/metadata.spec.ts

```diff
@@ -231,6 +232,50 @@ describe('modules/datasource/metadata', () => {
     ]);
   });
 
+  describe('massageUrl', () => {
+    it('Should return an empty string when massaging an invalid url', () => {
+      expect(massageGithubUrl('not a url')).toMatch('');
+    });
+
+    test.each`
+      sourceUrl
+      ${'git@github.com:user/repo'}
+      ${'http://github.com/user/repo'}
+      ${'http+git://github.com/user/repo'}
+      ${'https+git://github.com/user/repo'}
+      ${'ssh://git@github.com/user/repo'}
+      ${'git://github.com/user/repo'}
+      ${'https://www.github.com/user/repo'}
+      ${'https://user.github.com/repo'}
+    `('Should massage GitHub url $sourceUrl', ({ sourceUrl }) => {
+      expect(massageUrl(sourceUrl)).toBe('https://github.com/user/repo');
+    });
+
+    test.each`
+      sourceUrl
+      ${'http://gitlab.com/user/repo'}
+      ${'git://gitlab.com/user/repo'}
+      ${'https://gitlab.com/user/repo/tree/master'}
+      ${'http://gitlab.com/user/repo/'}
+      ${'http://gitlab.com/user/repo.git'}
+      ${'git@gitlab.com:user/repo.git'}
```

Comment:
what about gitlab ssh urls? we should do same as for github

Reply:
Same comment as above about the scope of the issue. I'm happy to add those URLs in another PR/issue.